plenum/server/replica.py

```diff
@@ -1950,7 +1955,7 @@ def __start_catchup_if_needed(self):
             # collections and process stashed messages which now fit between
             # watermarks
             self.caught_up_till_3pc((self.viewNo, stashed_checkpoint_ends[-1]))
-            self.processStashedMsgsForNewWaterMarks()
+            self.stasher.unstash_catchup()
```

Comment:
I'm not sure we should call any stasher method here.
Unstashing of watermarks needs to be done when watermarks are changed (setter for h?)

Reply:
Moved to setter for h